### ISSUE TITLE
[Chapter 4] Equation of Calculating Gradients in Stochastic Gradient Descent

### ISSUE THREAD

**Issue by MasudHaider** [ageron/handson-ml2#451]
Chapter 4 – Training Linear Models : Colab notebook cell 19
Calculating the gradients
      gradients = 2 * xi.T.dot(xi.dot(theta) - yi) 
shouldn't it be multiplied by 2/m instead of just 2 like the previous one in cell 16? Or am i missing something? Apology if, unnecessary issue was created. 

Thanks a ton

**Comment by ageron** [comment]
Hi @MasudHaider ,
The issue is closed so I guess you found the answer, but in case others are wondering, I'll answer anyway: in Stochastic Gradient Descent, at each iteration we approximate the gradient by computing it on a single training instance (instead of the whole training set as in Batch Gradient Descent). Since there's a single instance,  `m=1` in the equation, so `2 / m = 2`.
Hope this helps!

**Comment by MasudHaider** [comment]
I got it right after I opened the issue. That's why I closed it. Thanks for responding 😀
